src/nunavut/lang/c/__init__.py

```diff
@@ -207,11 +209,84 @@ def filter_macrofy(value: str) -> str:
         jinja_filter_tester(filter_macrofy, template, rendered, 'c')
 
 
+    .. code-block:: python
+
+        # Given
+        template = '#ifndef {{ "namespaced.Type.register" | macrofy }}'
+
+        # then
+        rendered = '#ifndef NAMESPACED_TYPE__REGISTER'
+
+        # Because register is a reserved word and is therefore stropped
+        # before being transformed.
+
+    .. invisible-code-block: python
+
+        jinja_filter_tester(filter_macrofy, template, rendered, 'c')
+
+    .. code-block:: python
+
+        # Given
+        template = '#ifndef {{ "_starts_with_underscore" | macrofy }}'
+
+        # then
+        rendered = '#ifndef _sTARTS_WITH_UNDERSCORE'
+
+        # Because the final token is stropped and must obey language
+        # rules.
+
+    .. invisible-code-block: python
+
+        jinja_filter_tester(filter_macrofy, template, rendered, 'c')
+
+        # Given
+        template = '#ifndef {{ "_invalid.register" | macrofy }}'
+
+        # then
+        rendered = '#ifndef _INVALID_REGISTER'
+
+        context = MagicMock(spec=LanguageContext)
+        language = MagicMock(spec=Language)
+        language.enable_stropping = False
+        context.get_language = MagicMock(return_value=language)
+        jinja_filter_tester(filter_macrofy, template, rendered, context)
+
+    .. code-block:: python
+
+        # You can disable stropping or override with the "partial_token" parameter
+        # which allows invalid identifiers to be generated.
+
+        # Given
+        template = '#ifndef FOO{{ "_invalid.register" | macrofy(partial_token=True) }}'
+
+        # then
+        rendered = '#ifndef FOO_INVALID__REGISTER'
+
+        # Stropping is still enabled so tokens are still stropped but the partial_token
+        # parameter skips stropping the final token.
```

Comment:
Double underscores cannot appear in user-defined identifiers. Maybe we should just replace them with a single underscore?